miso-web/src/main/webapp/scripts/list.js

```diff
@@ -23,81 +23,99 @@
 
 ListState = {};
 ListTarget = {};
-ListUtils = {
-  createTable : function(elementId, target, projectId, config) {
+ListUtils = (function() {
+  var initTable = function(elementId, target, projectId, config, optionModifier) {
     var searchKey = target.name + '_search';
     var lastSearch = window.localStorage.getItem(searchKey);
     var staticActions = target.createStaticActions(config, projectId);
     var bulkActions = target.createBulkActions(config, projectId);
     var columns = target.createColumns(config, projectId).filter(function(x) {
       return x.include;
     });
+    ListState[elementId] = {
+      selected : [],
+      data : [],
+      lastId : -1
+    };
     if (bulkActions.length > 0) {
-      ListState[elementId] = [];
-      columns.unshift(Utils.createToggleColumn('ListState.' + elementId));
+      columns
+          .unshift({
+            "sTitle" : "",
+            "mData" : "id",
+            "include" : true,
+            "mRender" : function(data, type, full) {
+              var checked = ListState[elementId].selected.some(function(obj) {
+                return obj.id == data;
+              }) ? " checked=\"checked\"" : "";
+              
+              return "<input type=\"checkbox\" id=\"" + elementId + "_toggle" + data + "\" onclick=\"ListUtils._checkEventHandler(this.checked, event, " + data + ", '" + elementId + "')\"" + checked + ">";
+            }
+          });
+      if (staticActions.length > 0) {
+        staticActions.push(null);
+      }
+      staticActions.push({
+        "name" : "☑",
+        "title" : "Select all item",
+        "handler" : function() {
+          var state = ListState[elementId];
+          state.lastId = -1;
+          state.selected = Utils.array.deduplicateById(state.selected
+              .concat(state.data));
+          state.data.forEach(function(item) {
+            var element = document
+                .getElementById(elementId + "_toggle" + item.id);
+            if (element) {
+              element.checked = true;
+            }
+          });
+        }
+      });
+      staticActions.push({
+        "name" : "☐",
+        "title" : "Clear selected items",
```

Comment:
I have a mild preference for `Deselect all` because `Select`/`Deselect` is more prominent than `all`/`none`, but it's alt text and I'd really be happy with either `Deselect all` or `Select none`.